### PR TITLE
Create setHtmlContentTypeOnError Property

### LINKED ISSUE #16118: PH34054 -create sethtmlcontenttypeonerror webcontainer property
A new webcontainer property is to be added: `sethtmlcontenttypeonerror`.
It stops the default content-type from being set when webcontainer handles errors. 

This property is likely most useful when creating APIs with Spring.

Values
- true -- (default)  sets  `text/html`
- false   -- sets nothing 
_____________

Background:

Fix for TS004280468

When an error occurs, the liberty server redirects the the error to a specific URL if specified by the web.xml. For example:

```
<error-page>
    <error-code>404</error-code>
    <location>/Error404.jsp</location>
</error-page>

```
If an application uses Spring, the content-type can be automatically determined (i.e considering the Accept header). However, liberty sets content-type first to html/text. Spring checks if any content-type headers exist before adding them (such as if it was added by the developer's servlet). The customer in this case needed application/xml or application/json to both set on different requests. 

Additionally, our servlet may be breaking the servlet spec. But this not exactly clear if it's handing errors because the webcontainer forwards the error message to the client (and it's typically text/html).

Spec: `Servlet containers must not set a default content type when the servlet programmer does not set the type.`

I wouldn't necessary say Spring is incorrect, but neither does it seem that we are either. However, desired behavior is not achieved nor consistent with other servers (Tomcat, for instance).

We'll create a new web container property to avoid setting the content type when specified.
__________

 Here's an example where the response returns an XmlRootElement annotated object 

` <webContainer setHtmlContentTypeOnError="true" /> (default)`

http://localhost:9080/PH34054/404 - > returns  application/xml

http://localhost:9080/PH34054/bad-request - > returns  text/html;charset=ISO-8859-1 <---  wrong   

......
` <webContainer setHtmlContentTypeOnError="false" />`

http://localhost:9080/PH34054/404 - > returns  application/xml

http://localhost:9080/PH34054/bad-request - > returns  application/xml   <---- Correct :)

### PR DESCRIPTION
**Pull request by volosied**
fixes #16118

### CODE REVIEW

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer.servlet.4.0_fat

File: dev/com.ibm.ws.webcontainer/src/com/ibm/wsspi/webcontainer/WCCustomProperties.java [hunk 1]
```diff
@@ -319,6 +319,9 @@
     //21.0.0.1
     public static boolean REDIRECT_TO_RELATIVE_URL;
 
+    //21.0.0.3
```
[the remark this answers is not in the document]
Comment (volosied) [reply dev/com.ibm.ws.webcontainer/src/com/ibm/wsspi/webcontainer/WCCustomProperties.java:322]: ah, yes I'll update

**Comment by volosied** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
#build

**Review by pmd1nh — APPROVED** [review]

**Comment by volosied** [comment]
#run-libby-bot

**Comment by volosied** [comment]
#run-libby-bot

**Pull request merged**